modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/DLViewMVCRenderCommand.java

```diff
@@ -101,6 +102,14 @@ public String render(
 						_portal.getHttpServletResponse(renderResponse),
 						dlAdminDisplayContext));
 
+			renderRequest.setAttribute(
+				DLWebKeys.DOCUMENT_LIBRARY_TRASH_HELPER, getDLTrashHelper());
+			renderRequest.setAttribute(
+				DLViewEntriesDisplayContext.class.getName(),
+				new DLViewEntriesDisplayContext(
+					_portal.getLiferayPortletRequest(renderRequest),
+					_portal.getLiferayPortletResponse(renderResponse)));
+
```

Comment:
Now that you're creating the display context here, why are you injecting the services in the request and getting them back in the constructor body? I think it would be much cleaner to inject the services directly into the constructor.